### ISSUE TITLE
$LATEST doesn't check that the partition suffix if a formatted string

### ISSUE THREAD

**Issue by martinbomio** [spotify/scio#764]
If I have two tables on the same dataset like:
```
project-id.dataset.my_table_20170808
project-id.dataset.my_table_v2_20170808
```
If my query has (project-id.dataset.my_table_%s, $LATEST)

The $LATEST will grab `project-id.dataset.my_table_v2_20170808` table even though the actual table name prefix has no v2.

The expected behaviour will to return the `project-id.dataset.my_table_20170808`.

**Comment by nevillelyh** [comment]
This is expected though since both have the prefix `project-id.dataset.my_table_` and the logic assumes all tables in the dataset follows the same partition scheme. Also note that `$LATEST` doesn't specify date format since it's more of a convention we use and not part of the BQ spec. 

It's better in this case to put v2 tables in a different dataset, make sense?

**Comment by martinbomio** [comment]
@nevillelyh yeah, that's what we are doing right now, but thought it could be open to discussion if smarter partition matching can be done in scio. 
Is it possible to make any assumption on the suffix?

**Comment by nevillelyh** [comment]
After discussion it's best to leave this out since we can't constraint the user's date format.

**Comment by nevillelyh** [comment]
Fixed in #781
